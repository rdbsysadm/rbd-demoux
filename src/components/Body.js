import PropTypes from "prop-types";
import React from "react";

import {
  ErrorBoundary,
  Facets,
  PagingInfo,
  Paging,
  Results,
  ResultsPerPage,
  Sorting
} from "../containers";
import { buildSortOptionsFromConfig } from "../config/config-helper";

function Body({ hasSidebar = true }) {
  return (
    <div className="reference-ui-body">
      <ErrorBoundary>
        <div className="initial-state-message">
          Type a search above to begin.
        </div>
        <div className="search-results">
          <div className={"sidebar" + (hasSidebar ? '' : ' hidden')}>
            <Sorting sortOptions={buildSortOptionsFromConfig()} />
            <Facets />
          </div>
          <div className="results">
            <div className="results__header">
              <div className="meta">
                <PagingInfo />
                <ResultsPerPage />
              </div>
            </div>
            <div className="results__body">
              <Results />
            </div>
            <div className="results__footer">
              <Paging />
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </div>
  );
}

Body.propTypes = {
  hasSidebar: PropTypes.bool
};

export default Body;
