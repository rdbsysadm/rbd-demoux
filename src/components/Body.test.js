import React from "react";
import Body from "./Body";
import { shallow } from "enzyme";

it("renders correctly with sidebar", () => {
  const wrapper = shallow(<Body hasSidebar={true}/>);
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly without the sidebar", () => {
  const wrapper = shallow(<Body hasSidebar={false} />);
  expect(wrapper).toMatchSnapshot();
});
