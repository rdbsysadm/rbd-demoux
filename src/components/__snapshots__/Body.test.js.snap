// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with sidebar 1`] = `
<div
  className="reference-ui-body"
>
  <Component>
    <div
      className="initial-state-message"
    >
      Type a search above to begin.
    </div>
    <div
      className="search-results"
    >
      <div
        className="sidebar"
      >
        <Component
          sortOptions={
            Array [
              Object {
                "direction": "",
                "name": "Relevance",
                "value": "",
              },
            ]
          }
        />
        <Component />
      </div>
      <div
        className="results"
      >
        <div
          className="results__header"
        >
          <div
            className="meta"
          >
            <Component />
            <Component />
          </div>
        </div>
        <div
          className="results__body"
        >
          <Component />
        </div>
        <div
          className="results__footer"
        >
          <Component />
        </div>
      </div>
    </div>
  </Component>
</div>
`;

exports[`renders correctly without the sidebar 1`] = `
<div
  className="reference-ui-body"
>
  <Component>
    <div
      className="initial-state-message"
    >
      Type a search above to begin.
    </div>
    <div
      className="search-results"
    >
      <div
        className="sidebar hidden"
      >
        <Component
          sortOptions={
            Array [
              Object {
                "direction": "",
                "name": "Relevance",
                "value": "",
              },
            ]
          }
        />
        <Component />
      </div>
      <div
        className="results"
      >
        <div
          className="results__header"
        >
          <div
            className="meta"
          >
            <Component />
            <Component />
          </div>
        </div>
        <div
          className="results__body"
        >
          <Component />
        </div>
        <div
          className="results__footer"
        >
          <Component />
        </div>
      </div>
    </div>
  </Component>
</div>
`;
